src/commonControls/numberControl/NumberControl.ts

```diff
@@ -550,850 +628,662 @@ export class NumberControl extends Control implements InteractionModelContributo
+        generator.addControlIntent(new GeneralControlIntent(), imData);
+        generator.addControlIntent(new ValueControlIntent(AmazonBuiltInSlotType.NUMBER), imData);
+        generator.addYesAndNoIntents();
 
-        this.initiativeFunc = this.confirmValue;
-        return true;
+        for (const [capability, actionSlotIds] of Object.entries(this.props.interactionModel.actions)) {
+            generator.ensureSlotValueIDsAreDefined(this.id, 'action', actionSlotIds);
+        }
+        generator.ensureSlotValueIDsAreDefined(this.id, 'target', this.props.interactionModel.targets);
     }
 
-    private confirmValue(input: ControlInput, resultBuilder: ControlResultBuilder): void {
-        this.state.lastInitiative.actName = ConfirmValueAct.name;
-        resultBuilder.addAct(
-            new ConfirmValueAct(this, {
-                value: this.state.value,
-                renderedValue:
-                    this.state.value !== undefined ? this.props.valueRenderer(this.state.value, input) : '',
-            }),
-        );
+    /**
+     * Directly set the value.
+     *
+     * @param value - Value, either an integer or a string that can be parsed as a integer.
+     */
+    setValue(value: string | number) {
+        this.state.value = typeof value === 'string' ? Number.parseInt(value, 10) : value;
     }
 
-    private isConfirmationRequired(input: ControlInput) {
-        if (typeof this.props.confirmationRequired === 'function') {
-            return this.props.confirmationRequired(this.state, input);
-        } else if (typeof this.props.confirmationRequired === 'boolean') {
-            return this.props.confirmationRequired;
-        } else {
-            return true; // by default confirmation is required
-        }
+    /**
+     * Clear the state of this control.
+     */
+    clear() {
+        this.state = new NumberControlState();
+        this.state.lastInitiative = {};
     }
 
-    private getAmbiguousPartner(value?: number): number | undefined {
-        const pairs = this.props.ambiguousPairs;
+    async renderAPLComponent(
+        input: ControlInput,
+        resultBuilder: ControlResponseBuilder,
+    ): Promise<{ [key: string]: any }> {
+        const aplRenderFunc = this.props.apl.renderComponent;
+        const defaultProps: NumberControlAPLComponentProps = {
+            validationFailedMessage: (value?: number) =>
+                i18next.t('NUMBER_CONTROL_DEFAULT_APL_INVALID_VALUE', { value }),
+            valueRenderer: this.props.valueRenderer,
+        };
+        return aplRenderFunc.call(this, this, defaultProps, input, resultBuilder);
+    }
 
-        for (const pair of pairs) {
-            if (pair[0] === value) {
-                return pair[1];
-            } else if (pair[1] === value) {
-                return pair[0];
-            }
-        }
-        return undefined;
+    private getAmbiguousPair(value: number, input: ControlInput): number {
```

Comment:
nit: name can be improved.

Reply:
Renamed to `getSuggestionForMisunderstandingValue`